src/main/java/com/red5pro/server/plugin/simpleauth/datasource/impl/roundtrip/stream/security/PlaybackSecurity.java

```diff
@@ -49,29 +54,42 @@
  * @author Rajdeep Rath
  *
  */
-
 public class PlaybackSecurity implements IStreamPlaybackSecurity {
 
 	private static Logger logger = LoggerFactory.getLogger(PlaybackSecurity.class);
 
-	private RoundTripAuthValidator roundTripAuthValidator;
+	private final RoundTripAuthValidator roundTripAuthValidator;
+
+	// default response for allowed or not
+	private boolean defaultResponse = true;
 
 	public PlaybackSecurity(RoundTripAuthValidator roundTripAuthValidator) {
 		this.roundTripAuthValidator = roundTripAuthValidator;
 	}
 
 	@Override
 	public boolean isPlaybackAllowed(IScope scope, String name, int start, int length, boolean flushPlaylist) {
+		// an npe is possible farther down, if the connection isn't available here
 		IConnection connection = Red5.getConnectionLocal();
-
-		Map<String, Object> attrs = connection.getAttributes();
-		Iterator<Map.Entry<String, Object>> it = attrs.entrySet().iterator();
-		while (it.hasNext()) {
-			Map.Entry<String, Object> pair = it.next();
-			logger.debug("key : {} value : {}", pair.getKey(), pair.getValue());
+		if (connection != null) {
+			// attrs inspection is only for debug level logging here, so we'll use a guard
+			// for optimization
+			if (logger.isDebugEnabled()) {
+				Map<String, Object> attrs = connection.getAttributes();
+				Iterator<Map.Entry<String, Object>> it = attrs.entrySet().iterator();
+				while (it.hasNext()) {
+					Map.Entry<String, Object> pair = it.next();
+					logger.debug("key : {} value : {}", pair.getKey(), pair.getValue());
+				}
+			}
+			return roundTripAuthValidator.onPlaybackAuthenticate(connection, scope, name);
 		}
+		// default playback result if a connection isn't present
+		return defaultResponse;
```

Comment:
@mondain What kind of scenario could result in a null connection that allows to playback, or publish in PublishSecurity.java, by default?

Reply:
A null connection can happen when connection local is improperly handled as well as when ConnectorShell is in-use (not 100% on that last one)